### PR TITLE
Fix a regression that selects popup menu items on mouse down.

### PR DESCRIPTION
**Pull request by ryevdokimov**
Selection should be on mouse up, not down as discussed here: https://github.com/godotengine/godot/issues/88324#issuecomment-1944670206

This does not fix the mentioned issue, just addresses that specific comment.

### CODE REVIEW

File: scene/gui/popup_menu.cpp [hunk 1]
```diff
@@ -588,8 +588,8 @@ void PopupMenu::_input_from_window_internal(const Ref<InputEvent> &p_event) {
 	}
 
 	if (m.is_valid() && drag_to_press) {
-		BitField<MouseButtonMask> initial_button_mask = m->get_button_mask();
-		if (!initial_button_mask.has_flag(mouse_button_to_mask(MouseButton::LEFT)) && !initial_button_mask.has_flag(mouse_button_to_mask(MouseButton::RIGHT))) {
+		DisplayServer *ds = DisplayServer::get_singleton();
+		if (!ds->mouse_get_button_state().has_flag(mouse_button_to_mask(MouseButton::LEFT)) && !ds->mouse_get_button_state().has_flag(mouse_button_to_mask(MouseButton::LEFT))) {
```
Comment (akien-mga) [inline scene/gui/popup_menu.cpp:592]: Copy paste mistake? It's checking `LEFT` twice.

Comment (ryevdokimov) [reply scene/gui/popup_menu.cpp]: Oops. I need a coffee.

**Comment by akien-mga** [comment]
Tested the current PR on Linux, that doesn't solve the issue for me.

Just right-clicking a folder in the FileSystem dock triggers the popup item that's immediately right of the cursor (which is on the immediate left edge of the PopupMenu).

[2024-02-15 14-17-43.webm](https://github.com/godotengine/godot/assets/4701338/69dd9ca7-70e4-4e33-becb-b21c885bb3d0)

**Comment by akien-mga** [comment]
Tested 9aba67b, but same issue on Linux/Wayland :/

**Comment by ryevdokimov** [comment]
I'm out of ideas then. Would need more feedback from someone with more understanding of the Inputs code or have someone with macOS/Linux help debug. 

On Windows it works whether I use DisplayServer or the passed InputEvent.

**Comment by akien-mga** [comment]
Superseded by #88392.

**Pull request closed without merging**